File: src/main/kotlin/com/ks/culinario/network/controller/ShoppingListController.kt
package com.ks.culinario.network.controller

import com.ks.culinario.domain.service.ShoppingListService
import com.ks.culinario.network.dto.NewShoppingListDTO
import com.ks.culinario.network.dto.ShoppingListDTO
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/shoppingList")
class ShoppingListController(private val shoppingListService: ShoppingListService) {

    @GetMapping("/all")
    fun getAll(): List<ShoppingListDTO> {
        return shoppingListService.getAll()
    }

    @GetMapping("/{id}")
    fun get(@PathVariable id: Int): ShoppingListDTO {
        return shoppingListService.get(id)
    }

    @PostMapping
    fun create(@RequestBody newShoppingListDTO: NewShoppingListDTO) {
         shoppingListService.create(newShoppingListDTO)
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Int) {
        shoppingListService.delete(id)
    }
}

File: dotnet/Culinario.API/Controllers/ShoppingListController.cs
using Culinario.Application.DTOs;
using Culinario.Application.Services;
using Microsoft.AspNetCore.Mvc;

namespace Culinario.API.Controllers;

[ApiController]
[Route("api/[controller]")]
public class ShoppingListController : ControllerBase
{
    private readonly IShoppingListService _shoppingListService;

    public ShoppingListController(IShoppingListService shoppingListService)
    {
        _shoppingListService = shoppingListService;
    }

    [HttpGet("all")]
    public async Task<ActionResult<List<ShoppingListDTO>>> GetAll()
    {
        return await _shoppingListService.GetAllAsync();
    }

    [HttpGet("{id}")]
    public async Task<ActionResult<ShoppingListDTO>> Get(int id)
    {
        try
        {
            return await _shoppingListService.GetAsync(id);
        }
        catch (Exception)
        {
            return NotFound();
        }
    }

    [HttpPost]
    public async Task<IActionResult> Create(NewShoppingListDTO dto)
    {
        await _shoppingListService.CreateAsync(dto);
        return Ok();
    }

    [HttpDelete("{id}")]
    public async Task<IActionResult> Delete(int id)
    {
        await _shoppingListService.DeleteAsync(id);
        return NoContent();
    }
}
